src/java.base/share/classes/java/nio/file/WatchService.java

```diff
@@ -98,6 +98,19 @@
  * it is not required that changes to files carried out on remote systems be
  * detected.
  *
+ * @implNote
+ * The JDK's {@code WatchService} implementations buffer up to 512 pending
+ * events for each registered watchable object. If this limit is exceeded,
+ * pending events are discarded and the special
+ * {@link StandardWatchEventKinds#OVERFLOW OVERFLOW} event is queued. This
+ * special event is the trigger to re-examine the state of the object, e.g.
+ * scan a watched directory to get an updated list of the files in the
+ * directory. The limit for the pending events can be changed from its default
+ * with the system property {@code jdk.nio.file.WatchService.maxEventsPerPoll}
```

Comment:
@fmeum On this line 109, would you please change `@code` to `@systemProperty`?

```
--- a/src/java.base/share/classes/java/nio/file/WatchService.java
+++ b/src/java.base/share/classes/java/nio/file/WatchService.java
@@ -106,7 +106,8 @@
  * special event is the trigger to re-examine the state of the object, e.g.
  * scan a watched directory to get an updated list of the files in the
  * directory. The limit for the pending events can be changed from its default
- * with the system property {@code jdk.nio.file.WatchService.maxEventsPerPoll}
+ * with the system property
+ * {@systemProperty jdk.nio.file.WatchService.maxEventsPerPoll}
  * set to a value that parses as a positive integer. This may be useful in
  * environments where there is a high volume of changes and where the impact
  * of discarded events is high.
  * ```

Reply:
Thanks, I didn't even know about this tag.

Reply:
Neither did I, until @AlanBateman made me aware of it.

I will run some more tests and file the CSR.

Thanks.